components/logins/src/sync/merge.rs

```diff
@@ -33,29 +33,56 @@ impl MirrorLogin {
         })
     }
 }
-
 #[derive(Clone, Debug)]
-pub(crate) struct LocalLogin {
-    pub login: EncryptedLogin,
-    pub local_modified: SystemTime,
+pub(crate) enum LocalLogin {
+    Tombstone {
+        id: String,
+        local_modified: SystemTime,
+    },
+    Alive {
+        login: EncryptedLogin,
+        local_modified: SystemTime,
+    },
 }
 
 impl LocalLogin {
     #[inline]
     pub fn guid_str(&self) -> &str {
-        self.login.guid_str()
+        match &self {
+            LocalLogin::Tombstone { id, .. } => id.as_str(),
+            LocalLogin::Alive { login, .. } => login.guid_str(),
+        }
+    }
+
+    pub fn local_modified(&self) -> SystemTime {
+        match &self {
+            LocalLogin::Tombstone { local_modified, .. }
+            | LocalLogin::Alive { local_modified, .. } => *local_modified,
+        }
     }
 
     pub(crate) fn from_row(row: &Row<'_>) -> Result<LocalLogin> {
-        Ok(LocalLogin {
-            login: EncryptedLogin::from_row(row)?,
-            local_modified: util::system_time_millis_from_row(row, "local_modified")?,
+        let local_modified = util::system_time_millis_from_row(row, "local_modified")?;
+        Ok(if row.get("is_deleted")? {
+            let id = row.get("guid")?;
+            LocalLogin::Tombstone { id, local_modified }
+        } else {
+            let login = EncryptedLogin::from_row(row)?;
+            if login.sec_fields.is_empty() {
+                error_support::report_error!("logins-crypto", "empty ciphertext in the db",);
+            }
```

Comment:
nice! This will most likely fail the sync if we hit it, Is it worth us returning an error as well report it?

Reply:
Good question! I was trying to remain conservative and not introduce new error conditions, but OTOH, the user is still going to have a bad time eventually - at decrypt time. So I'm leaning towards just doing this and let whatever existing failures might have happened continue to happen and re-evaluate once we start seeing sentry reports, but could easily be convinced to do something different instead.